test/extended/images/mongodb_ephemeral.go

```diff
@@ -39,22 +45,10 @@ var _ = g.Describe("[images][mongodb] openshift mongodb image", func() {
 			o.Expect(podNames).Should(o.HaveLen(1))
 
 			g.By("expecting the mongodb service is answering for ping")
-			mongo := db.NewMongoDB(podNames[0], "")
-
-			for times := 0; times < 10; times++ {
-				ok, err := mongo.IsReady(oc)
-				if ok {
-					break
-				}
-
-				if times == 10 {
-					o.Expect(err).ShouldNot(o.HaveOccurred())
-					o.Expect(ok).Should(o.BeTrue())
-					break
-				}
-
-				time.Sleep(1 * time.Second)
-			}
+			mongo := db.NewMongoDB(podNames[0])
+			ok, err := mongo.IsReady(oc)
```

Comment:
What `mongo.IsReady` does is to run `mongo --quiet --eval '{"ping", 1}'` on the first pod. Since the deployment is finished (`WaitForADeployment` above), the pod must be able to run that command at the first try. :sunny: 

Removing the retry loop is one of the benefits of using `WaitForADeployment`, that relies on API watchers.